pkg/resource/logging-config/alloy/logging.alloy.template

```diff
@@ -4,8 +4,9 @@ loki.source.podlogs "kubernetes_pods" {
   forward_to      = [loki.relabel.kubernetes_pods.receiver]
 
   selector {
-    match_labels = {
-      "giantswarm.io/managed-by" = "alloy-logs",
+    match_expression = {
+      key      = "giantswarm.io/managed-by"
```

Comment:
Why do we need this label? This means we cannot discover other pod lods right?

Reply:
Right, maybe we should keep it to `giantswarm.io/managed-by: alloy-logs` so it's clear those PodLogs are going through our alloy-logs app.

Reply:
But that means customers need to add giantswarm.io/managed-by so their pod logs are found right? Why not get all pod logs?

Reply:
Also, if we were to use a label we should use those https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#labels and not giantswarm custom ones

Reply:
Honestly, I think we're building limitations we don't need, or at least not yet. We discussed this morning that we did not want to go label based and now we need to add a label on the pod log anyway. I'm not sure I see value considering we most likely have no customer running alloy. @giantswarm/team-atlas what do you think? 

Also, if we were to use a label, we might as well use the same one we use for service monitors right?

Reply:
I removed this selector